### PR TITLE
Add support for 0.5.0 to GCP DM solution and default to the latest supported version.

### PR DESCRIPTION
**Pull request by selmanj**
/cc @salrashid123, @ayj

### CODE REVIEW

File: install/gcp/deployment_manager/istio-cluster.jinja [hunk 1]
```diff
@@ -97,7 +98,31 @@ resources:
...
+              ./install/kubernetes/webhook-create-signed-cert.sh \
+                  --service istio-sidecar-injector \
+                      --namespace istio-system \
+                      --secret sidecar-injector-certs
+              kubectl apply -f install/kubernetes/istio-sidecar-injector-configmap-release.yaml
+              cat install/kubernetes/istio-sidecar-injector.yaml | \
+                  ./install/kubernetes/webhook-patch-ca-bundle.sh > \
+                  install/kubernetes/istio-sidecar-injector-with-ca-bundle.yaml
+              kubectl apply -f install/kubernetes/istio-sidecar-injector-with-ca-bundle.yaml
+              kubectl label namespace default istio-injection=enabled
```
Comment (ayj) [inline install/gcp/deployment_manager/istio-cluster.jinja]: UX question - this only enables injection in the `default` namespace. Do we want it enabled cluster-wide? Or should instructions be updated to educate user that namespaces must be explicitly labeled to be subject to automatic injection?

**Comment by selmanj** [comment]
@ayj We should definitely call out the configuration in the getting started guide. I'll reach out to @salrashid123 who wrote most of the guide and send out an updated PR.

**Comment by salrashid123** [comment]
@ayj 
@selmanj 

I can file the PR to  update the docs.  
It sufficient to use the ```default``` namespace since we're necessarily always creating a new cluster for this (i.,e there's no way we'd be adding stuff to an existing cluster).

File: install/gcp/deployment_manager/istio-cluster.jinja [hunk 2]
```diff
@@ -8,7 +8,6 @@ resources:
     zone: {{ properties['zone'] }}
     cluster:
       name: {{ properties['gkeClusterName'] }}
-      enableKubernetesAlpha: true
       initialClusterVersion: 1.9.2-gke.0
```
Comment (salrashid123) [inline install/gcp/deployment_manager/istio-cluster.jinja]: Should we remove the ```initialClusterVersion``...it'll use the latest
GKE [zonal cluster](https://cloud.google.com/kubernetes-engine/versioning-and-upgrades#versions_available_for_new_cluster_masters)

File: install/gcp/deployment_manager/istio-cluster.jinja.schema [hunk 3]
```diff
@@ -43,11 +43,11 @@ properties:
   installIstioRelease:
     type: string
     description: Install Istio Release version.
-    default: 0.3.0
+    default: 0.5.0
     enum:
-      - 0.2.12
```
Comment (ldemailly) [inline install/gcp/deployment_manager/istio-cluster.jinja.schema]: keep that one and skip 0.3 and 0.4 maybe ?
also maybe wait for 0.5.1
(and broken record, those files don't belong here/in master referencing old versions _ I really think we should move them elsewhere)

File: install/gcp/deployment_manager/istio-cluster.jinja (hunk 2 above)
Comment (salrashid123) [reply install/gcp/deployment_manager/istio-cluster.jinja]: Nerverminds..the default zonal at the moment is ```1.7.12-gke.1``` on GKE so maybe explicitly set it here.  Once even zonal is default on 1.9+, we'll remove this

File: install/gcp/deployment_manager/istio-cluster.jinja (hunk 1 above)
Comment (salrashid123) [reply install/gcp/deployment_manager/istio-cluster.jinja]: sorry, i added in the response in separately below re ```default``` namespace.  We're creating the a new GKE cluster from scratch everythime so its fine to use the default.  I've got a PR i want to submit next week for the docs where i'll state that the script is only enabling auto injection for defaults

File: install/gcp/deployment_manager/istio-cluster.jinja [hunk 4]
```diff
@@ -97,7 +98,31 @@ resources:
           {% endif %}
 
           {% if  properties['enableAutomaticSidecarInjection'] %}
-            kubectl apply -f install/kubernetes/istio-initializer.yaml
+            {% if properties['installIstioRelease'] == "0.5.0" %}
+              # Setup automatic istio sidecar injection for the default namespace.
+              # Note that the original 0.5.0-tagged version of this script had
+	      # a timing bug; here we pull one commit later.
+              curl -s https://raw.githubusercontent.com/istio/istio/41203341818c4dada2ea5385cfedc7859c01e957/install/kubernetes/webhook-create-signed-cert.sh -o ./install/kubernetes/webhook-create-signed-cert.sh && chmod +x ./install/kubernetes/webhook-create-signed-cert.sh
```
Comment (salrashid123) [inline install/gcp/deployment_manager/istio-cluster.jinja]: Shouldn't these scripts come within the release .zip by default?  (its weird to download them out of band (not just here in this script but generally)

Istio's docs are missing calling this step [here](https://istio.io/docs/setup/kubernetes/sidecar-injection.html#installing-the-webhook).   I'll file an issue to include the script into the ```0.5.1``` release

Comment (costinm) [reply install/gcp/deployment_manager/istio-cluster.jinja]: They should be part of istioctl at some point. Brand new code, polish in progress.

File: install/gcp/deployment_manager/istio-cluster.jinja.schema (hunk 3 above)
Comment (costinm) [reply install/gcp/deployment_manager/istio-cluster.jinja.schema]: +1 on keeping 0.2.12 ( quarterly / stable ).

-1 on moving, for now it's good to have them all in one place, and ok to point to old versions.

File: install/gcp/deployment_manager/istio-cluster.jinja (hunk 4 above)
Comment (ldemailly) [reply install/gcp/deployment_manager/istio-cluster.jinja]: they are part of the release now but may have missed 0.5.0 but will be in 0.5.1 and 0.6.0

**Comment by selmanj** [comment]
Sorry for modifying this yet again; it was brought to my attention that the service graph functionality doesn't work for 0.5.0 (see https://github.com/istio/issues/issues/179). I added a quick sed fix to force usage of 0.4.0 for 0.5.0.

**Comment by selmanj** [comment]
Now that 0.5.1 is out let's hold off on this PR; I'll move directly to 0.5.1

**Comment by ldemailly** [comment]
#3462

**Pull request closed without merging**